app/client/components/payment/update/updatePaymentFlow.tsx

```diff
@@ -94,12 +97,22 @@ const PaymentMethodBar = (props: PaymentMethodProps) => (
 const subscriptionToPaymentMethod: (sub: Subscription) => PaymentMethod = (
   subscription: Subscription
 ) => {
-  if (subscription.card) {
+  if (subscription.paymentMethod === "Card" && subscription.card) {
     return PaymentMethod.card;
-  } else if (subscription.payPalEmail) {
+  } else if (
+    subscription.paymentMethod === "PayPal" &&
+    subscription.payPalEmail
+  ) {
     return PaymentMethod.payPal;
+  } else if (
+    subscription.paymentMethod === "DirectDebit" &&
```

Comment:
Won't this conflict with changes in https://github.com/guardian/manage-frontend/pull/128?

Reply:
https://github.com/guardian/manage-frontend/pull/128? actually branches off this, so should be fine, will rebase off master after this is merged if needs be. Thanks